domain/consensus/processes/syncmanager/blocklocator.go

```diff
@@ -57,6 +52,20 @@ func (sm *syncManager) createBlockLocator(lowHash, highHash *externalapi.DomainH
 		nextBlueScore := currentBlockBlueScore - step
 		if currentBlockBlueScore < step {
 			nextBlueScore = lowBlockGHOSTDAGData.BlueScore()
+		} else if nextBlueScore < lowBlockGHOSTDAGData.BlueScore() {
+			nextBlueScore = lowBlockGHOSTDAGData.BlueScore()
+		}
+
+		for currentBlockBlueScore-nextBlueScore >= sm.finalityDepth {
```

Comment:
Please move this logic to `dagTraversalManager.LowestChainBlockAboveOrEqualToBlueScore` and add a comment explaining what you did here